git browser: add menu item to change project directory

LinqLover opened the pull request:
Useful for repairing a project which's directory has been moved to another location on the disk. This does NOT move the directory on disk.

![squot-change-project-dir](https://user-images.githubusercontent.com/38782922/193898916-434df0b2-0e98-4784-bbef-5d022aed179f.png)

j4yk commented on src/Squit.package/SquitBrowser.class/instance/actionProjectChangeDirectory.st:
```diff
@@ -0,0 +1,11 @@
+		from: (FileDirectory on: gitRepository workingDir pathName).
+	directory ifNil: [^ self].
+	gitRepository workingDir: (FileSystem disk root resolve: directory pathName).
+	self projectSelection: self projectSelection. "updates repositoryExists, selectedHistorian, et al."
```
Ugh. I just watched a talk about creating the worst programming language ever. And now I am reminded how close the implementation behavior of the Git Browser is at times, in terms of not doing what you expect in the place where you expect it.

I would like to have the part that you want extracted from `projectIndex:` and call the extracted method instead.

LinqLover replied:
Done

j4yk commented on src/Squit.package/SquitBrowser.class/instance/projectListMenu.shifted..st:
```diff
@@ -0,0 +1,3 @@
+ui menu
+projectListMenu: aMenu shifted: aBoolean
+	^ self menu: aMenu for: #(menuProjectList menuProjectListShifted:) shifted: aBoolean
```
I assume that you have tried it out, but why is the second selector necessary? There is no pragma that takes the shifted argument here. Compare to `menuBranchListHook:` and `shiftedMenuBranchListHook:`. (Accordingly, I assume the non-shifted pragma selector could be removed from `branchListMenu:shifted:`...)

LinqLover replied:
No, you're totally right, no idea why I invented this. Fixed now :)

LinqLover commented:
@j4yk Whew, that has escaped my attention for a lot of time :grimacing: I think I fixed your critique. I also added another commit:

![image](https://github.com/hpi-swa/Squot/assets/38782922/30394839-3383-4d1f-a7b3-512fed1458e5)

LinqLover commented:
The failing  CI test seems not to be my fault ;)
